test/test_qat.py

```diff
@@ -129,9 +129,8 @@ def ref_op(x):
 
             # backward
             dout = torch.randn(result_ref.size(), dtype=torch.float)
-            result_actual.backward(dout, retain_graph=True)
-            grad_ref = input.grad.cpu()
             result_actual.backward(dout)
+            grad_ref = input.grad.cpu()
```

Comment:
Gradient calculation for the reference is missing. Suggest:
```python
result_ref = ref_op(input)
result_actual = qat_op(input)
dout = torch.randn(result_ref.size(), dtype=torch.float)
result_ref.backward(dout)
grad_ref = input.grad.cpu()
input.grad.data.zero_()
result_actual.backward(dout)
grad_actual = input.grad.cpu()
self.assertEqual(grad_ref, grad_actual)
```

w1.grad.data.zero_()
result_ref.backward(dout)